crates/worker/src/partition/state_machine/command_interpreter.rs

```diff
@@ -515,6 +641,91 @@ where
+                    // cancel uncompleted invocations
+                    EnrichedEntryHeader::Invoke {
+                        is_completed,
+                        resolution_result: Some(resolution_result),
+                    } if !is_completed => {
+                        let target_fid = FullInvocationId::new(
+                            resolution_result.service_name,
+                            resolution_result.service_key,
+                            resolution_result.invocation_uuid,
+                        );
+
+                        self.send_message(
+                            OutboxMessage::TerminateInvocation(InvocationTermination::cancel(
+                                target_fid,
+                            )),
+                            effects,
+                        );
+                    }
+                    EnrichedEntryHeader::Awakeable { is_completed }
+                    | EnrichedEntryHeader::GetState { is_completed }
+                    | EnrichedEntryHeader::Sleep { is_completed }
+                    | EnrichedEntryHeader::PollInputStream { is_completed }
+                        if !is_completed =>
+                    {
+                        match &invocation_status {
+                            InvocationStatusProjection::Invoked => {
+                                Self::handle_completion_for_invoked(
+                                    full_invocation_id.clone(),
+                                    Completion::new(journal_index, canceled_result.clone()),
+                                    effects,
+                                )
+                            }
+                            InvocationStatusProjection::Suspended(waiting_for_completed_entry) => {
+                                resume_invocation |= Self::handle_completion_for_suspended(
+                                    full_invocation_id.clone(),
+                                    Completion::new(journal_index, canceled_result.clone()),
+                                    waiting_for_completed_entry,
+                                    effects,
+                                );
+                            }
+                            InvocationStatusProjection::Virtual(notification_target) => {
+                                Self::handle_completion_for_virtual(
+                                    full_invocation_id.clone(),
+                                    Completion::new(journal_index, canceled_result.clone()),
+                                    notification_target.clone(),
+                                    effects,
+                                )
+                            }
+                        }
+                    }
+                    header => {
+                        assert!(
+                            header.is_completed().unwrap_or(true),
+                            "All non canceled journal entries must be completed."
+                        );
```

Comment:
debug_assert?

Reply:
I would like to keep it an assert so that it always fires if there is a violation.